workflow/rules/oncoprint.smk

```diff
@@ -1,24 +1,13 @@
-rule build_oncoprint_table:
-    input:
-        bcf=get_oncoprint_batch
-    output:
-        "results/plots/oncoprint/{batch}.{event}.tsv"
-    log:
-        "logs/oncoprint/{batch}.{event}.table.log"
-    conda:
-        "../envs/oncoprinttable.yaml"
-    script:
-        "../scripts/build_oncoprint_matrix.py"
-
 rule plot_oncoprint:
     input:
-        "results/plots/oncoprint/{batch}.{event}.tsv"
+        bcfs=get_oncoprint_batch
     output:
-        report("results/plots/oncoprint/{batch}.{event}.pdf", category="Oncoprint", caption="../report/oncoprint.rst")
+        report("results/plots/oncoprint/{batch}.{event}.html", category="Oncoprint", caption="../report/oncoprint.rst")
+    params: 
+        input=lambda wc, input: ["{name}={path}".format(name=bcf_path.split("/")[-1].split(".")[0], path=bcf_path) for bcf_path in input.bcfs]
```

Comment:
Instead of doing this, it is better to let the get_oncoprint_batch function return the group names only. Then, use that directly in params, and in input you write something like
```
lambda w: expand("results/merged-calls/{group}.{{event}}.fdr-controlled.bcf", group=get_oncoprint_batch(w))
```

Reply:
Done